scripts/lib/CIME/case/check_input_data.py

```diff
@@ -6,9 +6,45 @@
 from CIME.XML.inputdata import Inputdata
 import CIME.Servers
 
-import glob
+import glob, hashlib
 
 logger = logging.getLogger(__name__)
+# The inputdata_checksum.dat file will be read into this hash if it's available
+chksum_hash = dict()
+
+
+def _download_checksum_file(server, input_data_root):
+    """
+    Return True if successfully downloaded
+    """
+    rel_path = "../inputdata_checksum.dat"
+    full_path = os.path.join(input_data_root, "inputdata_checksum.dat")
+    protocol = type(server).__name__
+    if protocol == 'svn':
+        # svn server does not have this file
+        return False
+    logging.info("Trying to download file: '{}' to path '{}' using {} protocal.".format(rel_path, full_path, protocol))
+    tmpfile = None
+    if os.path.isfile(full_path):
+        tmpfile = full_path+".tmp"
+        os.rename(full_path, tmpfile)
+    # Use umask to make sure files are group read/writable. As long as parent directories
+    # have +s, then everything should work.
+    with SharedArea():
+        success = server.getfile(rel_path, full_path)
+        if success:
+            if tmpfile:
+                os.remove(tmpfile)
+        else:
+            if os.path.isfile(tmpfile):
+                os.rename(tmpfile, full_path)
+                logger.warning("Could not automatically download file "+full_path+
```

Comment:
Connected with some of my above comments: it seems like this warning is misleading for two reasons:

(1) Based on what you told me in person, it sounds like it's acceptable for some servers to not have this file, so maybe this shouldn't generate a warning? I wonder if each server's xml entry should say whether that file has a checksum or not; then we'll skip trying to download it if it doesn't have one, and just try to download it if it is supposed to have one (generating this warning if it isn't found).

(2) The comment about "Restoring existing version" is a bit misleading because it may be restoring a file for a completely different server right now.